storage/replica.go

```diff
@@ -1660,3 +1676,41 @@ func (r *Replica) maybeAddToSplitQueue() {
 		r.store.splitQueue.MaybeAdd(r, r.store.Clock().Now())
 	}
 }
+
+// processReplicaCmd get a raft command from command queue and
+// call processRaftCommand to execute in a loop until there is
+// not any command in the queue for 5 seconds.
+func (r *Replica) processReplicaCmd() {
+	for {
+		select {
+		case cmd := <-r.applyQueueChan:
+			err := r.processRaftCommand(cmd.idKey, cmd.index, cmd.raftCmd)
+			if cmd.callback != nil {
+				cmd.callback(err)
+			}
+
+		case <-r.store.stopper.ShouldStop():
+			return
+
+		default:
+			atomic.CompareAndSwapInt32(&r.hasProcessReplicaCmdGoroutine, 1, 0)
```

Comment:
super nit: this can be `atomicStoreInt32`. also, i would put this in a `defer` because it's more obviously correct.